pkgs/os-specific/darwin/apple-source-releases/system_cmds/default.nix

```diff
@@ -1,114 +1,97 @@
-{ stdenv, appleDerivation, lib
-, libutil, Librpcsvc, apple_sdk, pam, CF, openbsm }:
+{
+  lib,
+  fetchFromGitHub,
+  appleDerivation,
+  libutil,
+  ncurses,
+  openbsm,
+  pam,
+  Librpcsvc,
+  xnu,
+  CoreFoundation,
+  CoreSymbolication,
+  DirectoryService,
+  IOKit,
+  Kernel,
+  OpenDirectory,
+  stdenv,
+  gawk,
+  meson,
+  ninja,
+  pkg-config,
+}:
 
-appleDerivation {
-  # xcbuild fails with:
-  # /nix/store/fc0rz62dh8vr648qi7hnqyik6zi5sqx8-xcbuild-wrapper/nix-support/setup-hook: line 1:  9083 Segmentation fault: 11  xcodebuild OTHER_CFLAGS="$NIX_CFLAGS_COMPILE" OTHER_CPLUSPLUSFLAGS="$NIX_CFLAGS_COMPILE" OTHER_LDFLAGS="$NIX_LDFLAGS" build
-  # see issue facebook/xcbuild#188
-  # buildInputs = [ xcbuild ];
+let
+  OpenDirectoryOSS.src = fetchFromGitHub {
+    owner = "apple-oss-distributions";
+    repo = "OpenDirectory";
+    rev = "OpenDirectory-146";
+    hash = "sha256-6fSl8PasCZSBfe0ftaePcBuSEO3syb6kK+mfDI6iR7A=";
+  };
+in
+appleDerivation (finalAttrs: {
+  postUnpack = ''
+    mkdir -p "$sourceRoot/CFOpenDirectory" "$sourceRoot/OpenDirectory"
+    install -t "$sourceRoot/CFOpenDirectory" \
+      '${OpenDirectoryOSS.src}/Core/CFOpenDirectoryPriv.h' \
+      '${OpenDirectoryOSS.src}/Core/CFODTrigger.h'
+    touch "$sourceRoot/CFOpenDirectory/CFOpenDirectoryConstantsPriv.h"
+    install -t "$sourceRoot/OpenDirectory" \
+      '${OpenDirectoryOSS.src}/Framework/OpenDirectoryPriv.h' \
+      '${OpenDirectoryOSS.src}/Framework/NSOpenDirectoryPriv.h'
+  '';
 
-  buildInputs = [ libutil Librpcsvc apple_sdk.frameworks.OpenDirectory pam CF
-                  apple_sdk.frameworks.IOKit openbsm ];
-  # env.NIX_CFLAGS_COMPILE = lib.optionalString hostPlatform.isi686 "-D__i386__"
-  #                    + lib.optionalString hostPlatform.isx86_64 "-D__x86_64__"
-  #                    + lib.optionalString hostPlatform.isAarch32 "-D__arm__";
-  env.NIX_CFLAGS_COMPILE = toString ([ "-DDAEMON_UID=1"
-                         "-DDAEMON_GID=1"
-                         "-DDEFAULT_AT_QUEUE='a'"
-                         "-DDEFAULT_BATCH_QUEUE='b'"
-                         "-DPERM_PATH=\"/usr/lib/cron/\""
-                         "-DOPEN_DIRECTORY"
-                         "-DNO_DIRECT_RPC"
-                         "-DAPPLE_GETCONF_UNDERSCORE"
-                         "-DAPPLE_GETCONF_SPEC"
-                         "-DUSE_PAM"
-                         "-DUSE_BSM_AUDIT"
-                         "-D_PW_NAME_LEN=MAXLOGNAME"
-                         "-D_PW_YPTOKEN=\"__YP!\""
-                         "-DAHZV1=64 "
-                         "-DAU_SESSION_FLAG_HAS_TTY=0x4000"
-                         "-DAU_SESSION_FLAG_HAS_AUTHENTICATED=0x4000"
-                       ] ++ lib.optional (!stdenv.isLinux) " -D__FreeBSD__ ");
+  env = {
+    NIX_CFLAGS_COMPILE = "-iframework ${OpenDirectory}/Library/Frameworks/OpenDirectory.framework/Versions/A/Frameworks";
+  };
```

Comment:
can't this get passed via a meson flag?

`mesonFlags = [ "CFLAGS=xxx" ];`
or `env.CFLAGS=`. `NIX_CFLAGS_COMPILE` seems like a last resort with a nonconforming build system.

Reply:
I’ll have to check in the morning, but I’m pretty sure the reason I did this is because frameworks in build inputs are normally added automatically to `NIX_CFLAGS_COMPILE`, but this is an embedded framework. The `meson.build` still does a check for the framework and links it as necessary.

Reply:
I refactored the derivation to pass the paths needed for private headers and added them as dependencies in `meson.build`.